### PR TITLE
[SRVLOGIC-287] Recreate Operator/Bundle files dropped from upstream in midstream

### PR DESCRIPTION
**Pull request by rgdoliveira**
**Description of the change:**

**Motivation for the change:**

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by mareknovotny — APPROVED** [review]

File: images/prod-bundle-image.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ modules:
   repositories:
     - path: modules
   install:
-    - name: org.kie.kogito.osl.bundleinstall
+    - name: org.kie.kogito.prod.bundleinstall
```
Comment (mareknovotny) [inline images/prod-bundle-image.yaml]: not sure if this is appropriate name, i would remove kogito and use maybe sonataflow and do not add prod?
@wmedvede @ricardozanini wdyt? I know this is recreation of original bundle, but it looks to me weird to use `kogito.prod.`

Comment (rgdoliveira) [reply images/prod-bundle-image.yaml]: TBH what I did was to replace all `osl` string by `prod` to try to change the minimum possible. But up to you guys...

Comment (ricardozanini) [reply images/prod-bundle-image.yaml]: @rgdoliveira can you use the same namespace I added in https://github.com/kiegroup/kogito-serverless-operator/pull/33/files#diff-809dd23de5c93bf2d838799633b8b4f3330d846cf502de15163387a1663e03d0R22

`com.redhat.osl`

Comment (rgdoliveira) [reply images/prod-bundle-image.yaml]: Done, I've changed the modules name and folder to use `com.redhat.osl`

File: images/prod-operator-image.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- schema_version: 1
+  name: "operator-builder"
+  version: "1.32.0"
+  from: "registry.access.redhat.com/ubi8/go-toolset:1.20.12"
```
Comment (rgdoliveira) [inline images/prod-operator-image.yaml:18]: @ricardozanini as we talked in the meeting today, I bumped to 1.20.12

File: bundle.prod/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+apiVersion: v1
+data:
+  DEFAULT_BUILDER_RESOURCE_NAME: Dockerfile
```
Comment (ricardozanini) [inline bundle.prod/manifests/logic-operator-rhel8-builder-config_v1_configmap.yaml:3]: I think I've removed this line, but we can fix this if after this PR gets merged. Ideally, the `logic-*` files should be copied from the upstream part and apply a template overriding prod unique info.

**Review by ricardozanini — APPROVED** [review]

**Review by wmedvede — APPROVED** [review]

**Comment by wmedvede** [comment]
@rgdoliveira e2e failing test might be related with the fact that the swf-builder image is now updated to quarkus 3.8.4.
And to handle this upate I have created this PR: #42 .

So, maybe we need to:
1) Merge  #42  when green.
2) Rebase this PR with main and see if we get the e2e greens.

Current persistence related failure.

Summarizing 1 Failure:
  [FAIL] Validate the persistence  when deploying a SonataFlow CR with PostgreSQL persistence [It] defined in the workflow from an existing kubernetes service as a reference

**Comment by wmedvede** [comment]
@rgdoliveira greens!

**Pull request merged**